src/main/java/org/opensearch/performanceanalyzer/commons/os/ThreadSched.java

```diff
@@ -0,0 +1,163 @@
+        public final double avgWaittime;
+        public final double contextSwitchRate; // both voluntary and involuntary
+
+        SchedMetrics(double avgRuntime, double avgWaittime, double contextSwitchRate) {
+            this.avgRuntime = avgRuntime;
+            this.avgWaittime = avgWaittime;
+            this.contextSwitchRate = contextSwitchRate;
+        }
+
+        @Override
+        public String toString() {
+            return new StringBuilder()
+                    .append("avgruntime: ")
+                    .append(avgRuntime)
+                    .append(" avgwaittime: ")
+                    .append(avgWaittime)
+                    .append(" ctxrate: ")
+                    .append(contextSwitchRate)
+                    .toString();
+        }
+    }
+
+    private LinuxSchedMetricsGenerator schedLatencyMap = new LinuxSchedMetricsGenerator();
+
+    private static String[] schedKeys = {"runticks", "waitticks", "totctxsws"};
+
+    private static SchemaFileParser.FieldTypes[] schedTypes = {
+        SchemaFileParser.FieldTypes.ULONG,
+        SchemaFileParser.FieldTypes.ULONG,
+        SchemaFileParser.FieldTypes.ULONG
+    };
+
+    private ThreadSched() {
+        try {
+            pid = OSGlobals.getPid();
+            tids = OSGlobals.getTids();
+        } catch (Exception e) {
+            LOGGER.error(
+                    (Supplier<?>)
+                            () ->
+                                    new ParameterizedMessage(
+                                            "Error In Initializing ThreadCPU: {}", e.toString()),
+                    e);
+        }
+    }
+
+    public synchronized void addSample() {
+        tids = OSGlobals.getTids();
+
+        oldtidKVMap.clear();
+        oldtidKVMap.putAll(tidKVMap);
+
+        tidKVMap.clear();
+        oldkvTimestamp = kvTimestamp;
+        kvTimestamp = System.currentTimeMillis();
+        for (String tid : tids) {
+            Map<String, Object> sample =
+                    (new SchemaFileParser(
+                                    "/proc/" + pid + "/task/" + tid + "/schedstat",
+                                    schedKeys,
+                                    schedTypes))
+                            .parse();
+            tidKVMap.put(tid, sample);
+        }
+
+        calculateSchedLatency();
+    }
+
+    public synchronized void addSample(String threadId) {
+        tids = OSGlobals.getTids();
+
+        oldtidKVMap.clear();
+        oldtidKVMap.putAll(tidKVMap);
+
+        tidKVMap.clear();
+        oldkvTimestamp = kvTimestamp;
+        kvTimestamp = System.currentTimeMillis();
+        Map<String, Object> sample =
+                (new SchemaFileParser(
+                                "/proc/" + pid + "/task/" + threadId + "/schedstat",
+                                schedKeys,
+                                schedTypes))
+                        .parse();
+        tidKVMap.put(threadId, sample);
+
+        calculateSchedLatency();
+    }
```

Comment:
Let us not repeat code, can we refactor this to:
```
private static final ALL_THREADS = "ALL_THREADS";

public synchronized void addSampleForThread(String threadId) {
        addSample(threadId);
}

public synchronized void addSample(String threadId = ALL_THREADS) {
        ...
}
```

Reply:
Hm I am afraid we can't provide default value for the method parameter in Java (like we can do in Kotlin for example).